dev/devicelab/bin/tasks/build_aar_module_test.dart

```diff
@@ -0,0 +1,225 @@
+        repoPath,
+        'io',
+        'flutter',
+        'devicelab',
+        'hello',
+        'flutter_release',
+        '1.0',
+        'flutter_release-1.0.pom',
+      ));
+
+      checkFileExists(path.join(
+        repoPath,
+        'io',
+        'flutter',
+        'plugins',
+        'deviceinfo',
+        'device_info_release',
+        '1.0',
+        'device_info_release-1.0.aar',
+      ));
+
+      checkFileExists(path.join(
+        repoPath,
+        'io',
+        'flutter',
+        'plugins',
+        'deviceinfo',
+        'device_info_release',
+        '1.0',
+        'device_info_release-1.0.pom',
+      ));
+
+      checkFileExists(path.join(
+        repoPath,
+        'io',
+        'flutter',
+        'plugins',
+        'packageinfo',
+        'package_info_release',
+        '1.0',
+        'package_info_release-1.0.aar',
+      ));
+
+      checkFileExists(path.join(
+        repoPath,
+        'io',
+        'flutter',
+        'plugins',
+        'packageinfo',
+        'package_info_release',
+        '1.0',
+        'package_info_release-1.0.pom',
+      ));
+
+      section('Build debug AAR');
+
+      await inDirectory(projectDir, () async {
+        await flutter(
+          'build',
+          options: <String>['aar', '--verbose', '--debug'],
+        );
+      });
+
+      checkFileExists(path.join(
+        repoPath,
+        'io',
+        'flutter',
+        'devicelab',
+        'hello',
+        'flutter_release',
+        '1.0',
+        'flutter_release-1.0.aar',
+      ));
+
+      checkFileExists(path.join(
+        repoPath,
+        'io',
+        'flutter',
+        'devicelab',
+        'hello',
+        'flutter_debug',
+        '1.0',
+        'flutter_debug-1.0.pom',
+      ));
+
+      checkFileExists(path.join(
+        repoPath,
+        'io',
+        'flutter',
+        'plugins',
+        'deviceinfo',
+        'device_info_debug',
+        '1.0',
+        'device_info_debug-1.0.aar',
+      ));
+
+      checkFileExists(path.join(
+        repoPath,
+        'io',
+        'flutter',
+        'plugins',
+        'deviceinfo',
+        'device_info_debug',
+        '1.0',
+        'device_info_debug-1.0.pom',
+      ));
+
+      checkFileExists(path.join(
+        repoPath,
+        'io',
+        'flutter',
+        'plugins',
+        'packageinfo',
+        'package_info_debug',
+        '1.0',
+        'package_info_debug-1.0.aar',
+      ));
+
+      checkFileExists(path.join(
+        repoPath,
+        'io',
+        'flutter',
+        'plugins',
+        'packageinfo',
+        'package_info_debug',
+        '1.0',
+        'package_info_debug-1.0.pom',
+      ));
+
+      return TaskResult.success(null);
+    } catch (e) {
+      return TaskResult.failure(e.toString());
+    } finally {
+      rmTree(tempDir);
+    }
+  });
+}
+
+void checkFileExists(String file) {
+  if (!exists(File(file))) {
+    throw 'Expected file `$file` to exit.';
```

Comment:
nit: generally best practice is to throw Exception or Error and not arbitrary objects.

Reply:
I think you want something like `throw FileSystemException('Expected file to exist', file)`

Reply:
Thanks.  I agree using an exception object is a better practice that arbitrary objects.

I initially used strings because there's a catch block in the code that calls `toString()`  on the exception to transform it into a return `TaskResult.failure`, so from this perspective the exception object doesn't add much.